SparkleSpin/Controllers/PlayerViewController.swift

```diff
@@ -11,51 +11,97 @@ import UIKit
+        navigationItem.title = "Add Players!"
+    }
+    
+    private func addDoneBarButtonItem() {
+        doneBarButton = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: nil)
+        navigationItem.rightBarButtonItem = doneBarButton
+        doneBarButton?.isEnabled = false
     }
     
     private func setupTableView() {
-        playerTableView.dataSource = self
+        playerTableView.dataSource = playerViewModel
         playerTableView.delegate = self
         playerTableView.backgroundColor = ThemeColor.Light.primaryColor
         playerTableView.separatorStyle = .none
         playerTableView.rowHeight = UITableView.automaticDimension
         playerTableView.estimatedRowHeight = CGFloat(Constants.estimatedRowHeight)
         playerTableView.tableFooterView = UIView()
     }
-
-}
-
-extension PlayerViewController: UITableViewDataSource, UITableViewDelegate {
     
-    func registerNib() {
+    private func registerNib() {
         let entryCell = UINib(nibName: NibID.entryCell, bundle: nil)
         playerTableView.register(entryCell, forCellReuseIdentifier: CellID.entryCell)
     }
     
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return 1
+    private func addPlayerToList() {
+        let nameEntry = playerEntryTextField.text ?? ""
+        var playerToAdd = playerViewModel.createPlayerWith(name: nameEntry)
+        playerViewModel.addPlayerToSavedList(player: &playerToAdd)
+        playerEntryTextField.text? = ""
+    }
+    
+    private func updateDoneBarButton() {
+        if doneBarButton?.isEnabled == false {
+            doneBarButton?.isEnabled = true
+        }
+    }
+    
+    private func insertNewPlayerRowInTable() {
+        playerTableView.beginUpdates()
+        let nextRowIndexPath = IndexPath(row: playerViewModel.playerList.count - 1, section: 0)
+        let nextCell = playerTableView.cellForRow(at: nextRowIndexPath) as? EntryCell
+        playerTableView.insertRows(at: [nextRowIndexPath], with: .bottom)
+        nextCell?.setCellStateWith(state: .saved)
+        playerTableView.endUpdates()
     }
     
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        guard let entryCell = tableView.dequeueReusableCell(withIdentifier: CellID.entryCell, for: indexPath) as? EntryCell else { fatalError("Fatal error: No cell") }
-        return entryCell
+    private func alertForEmptyEntry() {
+        let emptyEntryAlert = UIAlertController(title: "Oops!", message: "Entry cannot be blank!", preferredStyle: .alert)
+        emptyEntryAlert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+        present(emptyEntryAlert, animated: true)
     }
     
+    @IBAction func userTappedAddButton(_ sender: LightButton) {
+        addPlayerButton.animateButton()
+        guard let playerEntryText = playerEntryTextField.text else { return }
```

Comment:
With this in place looks like we can still add empty spaces as the name. Maybe we guard for that too
![empty](https://user-images.githubusercontent.com/5129887/55846743-d3a83300-5b14-11e9-9111-c13e0c733a09.gif)

Reply:
Good catch! Fixed in https://github.com/BritneyS/SparkleSpin/pull/10/commits/2528115cddec1cde2758bfa6cb279898b88d753f